qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -749,7 +760,8 @@ def _get_status_position(
                 if queue_info._status == ApiJobStatus.PENDING_IN_QUEUE.value:
                     status = JobStatus.QUEUED
         except (KeyError, ValidationError) as ex:
-            raise IBMQJobApiError("Unexpected return value received from the server.") from ex
+            raise IBMQJobApiError('Unexpected return value received from the server when '
+                                  'getting status for job {}: {}'.format(self.job_id(), str(ex)))
```

Comment:
why get rid of `from ex`? it seems more helpful to say exception A was caused by B.